### PR TITLE
#167484547 Users should be presented with a reset password

### PR DESCRIPTION
**Pull request by raymond42**
### What does this PR do?
- Users should be presented with a reset password
### Description of Task to be completed?
- Implement forgot password functionalities
- Implement reset password functionalities
### How should this be manually tested?
- Clone this branch
- `npm install` to install all dependencies
- add this environment variable in your .env file where you are running your backend `FRONTEND_URL = http://localhost:3000`
    N.B. you will use localhost PORT according to the port you are using for your frontend app which in this case for me it is 3000
- `npm start` to run the server for the frontend
- `npm start` to run the server for the backend
- Use this route `http://localhost:3000/forgot` to go to the login page
- N.B Provide a valid email
- Check your email and use the link sent to you to reset your password
- Click on the link and get to the reset password page

### What are the relevant pivotal tracker stories?
- [changing the reset password link to fronted url](https://www.pivotaltracker.com/story/show/169040218)

### Screenshots
Screen Shot 2019-10-10 at 03 55 20

Screen Shot 2019-10-10 at 04 29 48
Screen Shot 2019-10-10 at 03 52 27
Screen Shot 2019-10-10 at 04 29 05
Screen Shot 2019-10-10 at 03 53 01

### CODE REVIEW

**Review by placideirandora — CHANGES_REQUESTED** [review]
1. I checked and found out that you didn't write end to end tests for your feature, password reset. You should write some.

2. I think you should increase the size of the input bottom border to match the one of the login and signup.

3. I think you should also increase the size of the placeholder text to match the one of the login and signup.

Screen Shot 2019-10-10 at 20 41 47

4.  I think you should align the signup button with the other elements for consistency.
Screen Shot 2019-10-10 at 20 45 13

5. I think you should place the content in the center.
Screen Shot 2019-10-10 at 20 46 12

6. I think you should improve the responsiveness of smartphones. I have checked with: Galaxy S5, Pixel 2, and iPhone 4 (and all iPhone versions).
Screen Shot 2019-10-10 at 20 49 17
Screen Shot 2019-10-10 at 20 49 07
Screen Shot 2019-10-10 at 20 49 27

7. You should also improve the responsiveness of tablets. The size should decrease to reflect on the screen width. As you can see, the button text 'Reset Password' has wrapped. It should be on the same line.
Screen Shot 2019-10-10 at 20 52 01

8. You should fix the following bug: When I try to reset my password, I get the message: Network error.
Screen Shot 2019-10-10 at 20 57 55

**Comment by raymond42** [comment]
> 1. I checked and found out that you didn't write end to end tests for your feature, password reset. You should write some.
> 2. I think you should increase the size of the input bottom border to match the one of the login and signup.
> 3. I think you should also increase the size of the placeholder text to match the one of the login and signup.
> 
> Screen Shot 2019-10-10 at 20 41 47
> 
> 1. I think you should align the signup button with the other elements for consistency.
> 
> Screen Shot 2019-10-10 at 20 45 13
> 
> 1. I think you should place the content in the center.
> 
> Screen Shot 2019-10-10 at 20 46 12
> 
> 1. I think you should improve the responsiveness of smartphones. I have checked with: Galaxy S5, Pixel 2, and iPhone 4 (and all iPhone versions).
> 
> Screen Shot 2019-10-10 at 20 49 17
> 
> Screen Shot 2019-10-10 at 20 49 07
> 
> Screen Shot 2019-10-10 at 20 49 27
> 
> 1. You should also improve the responsiveness of tablets. The size should decrease to reflect on the screen width. As you can see, the button text 'Reset Password' has wrapped. It should be on the same line.
> 
> Screen Shot 2019-10-10 at 20 52 01
> 
> 1. You should fix the following bug: When I try to reset my password, I get the message: Network error.
> 
> Screen Shot 2019-10-10 at 20 57 55

@placideirandora I implemented your feedback. Thank you

**Review by placideirandora — CHANGES_REQUESTED** [review]
1. After getting the password reset link and clicked on it, I didn't get a page for resetting my password, instead, I was presented with an empty page. You should fix this.
Screen Shot 2019-10-11 at 10 46 55
Screen Shot 2019-10-11 at 10 46 43

**Comment by raymond42** [comment]
> 1. After getting the password reset link and clicked on it, I didn't get a page for resetting my password, instead, I was presented with an empty page. You should fix this.
> 
> Screen Shot 2019-10-11 at 10 46 55
> 
> Screen Shot 2019-10-11 at 10 46 43

Thanks. fixed

**Review by harerakalex — APPROVED** [review]
LGTM

File: src/feature/Reset Password/forgot password/ForgotPasswordComponent.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+export class ForgotPassword extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+    };
+  }
+
+  onChange = (e) => {
```
Comment (clemwek) [inline src/feature/Reset Password/forgot password/ForgotPasswordComponent.js]: Use `event` in full.

File: src/feature/Reset Password/forgot password/ForgotPasswordComponent.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  onSubmit = (e) => {
```
Comment (clemwek) [inline src/feature/Reset Password/forgot password/ForgotPasswordComponent.js]: Use `event` in full

**Review by clemwek — APPROVED** [review]
Fix the small changes then merge

File: src/feature/Reset Password/forgot password/ForgotPasswordComponent.js (hunk 1 above)
Comment (raymond42) [reply src/feature/Reset Password/forgot password/ForgotPasswordComponent.js]: on it

File: src/feature/Reset Password/forgot password/ForgotPasswordComponent.js (hunk 2 above)
Comment (raymond42) [reply src/feature/Reset Password/forgot password/ForgotPasswordComponent.js]: on it

**Pull request merged**